builtin/providers/nsone/resource_record.go

```diff
@@ -0,0 +1,854 @@
+							Optional: true,
+						},
+						"meta": &schema.Schema{
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"field": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"feed": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+										//ConflictsWith: []string{"value"},
+									},
+									"value": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+										//ConflictsWith: []string{"feed"},
+									},
+								},
+							},
+							Set: metaToHash,
+						},
+					},
+				},
+				Set: answersToHash,
+			},
+			"regions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"georegion": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, es []error) {
+								value := v.(string)
+								if !regexp.MustCompile(`^(US-WEST|US-EAST|US-CENTRAL|EUROPE|AFRICA|ASIAPAC|SOUTH-AMERICA)$`).MatchString(value) {
+									es = append(es, fmt.Errorf(
+										"only US-WEST, US-EAST, US-CENTRAL, EUROPE, AFRICA, ASIAPAC, SOUTH-AMERICA allowed in %q", k))
+								}
+								return
+							},
+						},
+						"country": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"us_state": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"up": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+				Set: regionsToHash,
+			},
+			"filters": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"filter": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"disabled": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"config": &schema.Schema{
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+					},
+				},
+			},
+			// Computed
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+		Create: RecordCreate,
+		Read:   RecordRead,
+		Update: RecordUpdate,
+		Delete: RecordDelete,
+	}
+}
+
+func regionsToHash(v interface{}) int {
+	var buf bytes.Buffer
+	r := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", r["name"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", r["georegion"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", r["country"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", r["us_state"].(string)))
+	buf.WriteString(fmt.Sprintf("%t-", r["up"].(bool)))
```

Comment:
Each region is a name mapping to the ns1 metadata structure. so we should probably be hashing every possible field in the ns1 metadata [reference](https://github.com/hashicorp/terraform/pull/10058/files#diff-5edfdd3df9b3556a76a1495ea19af3a0R20). We have a note in our api docs about how the name `regions` should really be `groups`, i think that really helps clear up what the field contains. 

[From ns1 api docs](https://ns1.com/api/#record-put)
> Metadata tables (meta) may be specified in answers, in regions or in the record as a whole. (Note that regions can be any kind of group, not just geo-related. For this reason, the NS1 web portal at my.nsone.net uses the word "group" instead of "region".) The metadata tables may contain key/value pairs where valid keys and values are as described in the output of /metatypes.

> Anywhere metadata tables can go, data feeds can go as well. Connect a compatible feed to a metadata key by assigning its value to be a JSON object pointing to the data feed that should be consulted for the final value.

We also have different types for these fields, based on whether the value is static or dynamic(a datafeed id)